Decred Wallet/Features/Navigation Menu/NavigationMenuTabBarController.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  NavigationTabBarController.swift
+//  Decred Wallet
+//
+// Copyright (c) 2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+import UIKit
+
+class NavigationMenuTabBarController: UITabBarController {
+    
+    var isNewWallet: Bool = false
+    var customTabBar: CustomTabMenuView!
+    lazy var floatingButtons: NavMenuFloatingButtons = {
+        return NavMenuFloatingButtons()
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.loadTabBar()
+        if self.isNewWallet {
+            self.showOkAlert(message: LocalizedStrings.newWalletMsg)
+        }
+    }
+    
+    func loadTabBar() {
+        let tabItems: [MenuItem] = [.overview, .transactions, .wallets, .more]
+        self.setupCustomTabMenu(tabItems)
+        self.selectedIndex = 0
+    }
+    
+    // Create our custom menu bar and display it right where the tab bar should be.
+    func setupCustomTabMenu(_ menuItems: [MenuItem]) {
+        tabBar.isHidden = true
+        self.customTabBar = CustomTabMenuView(menuItems: menuItems, frame: tabBar.frame) // Draw and layout the tab navigation menu
+        
+        self.view.addSubview(customTabBar)
+        self.customTabBar.translatesAutoresizingMaskIntoConstraints = false // We are setting positioning constraints in the next line. best to ignore XCode generated constraints
+        self.customTabBar.clipsToBounds = true
+        
+        // Positioning constraints to place the nav menu right where the tab bar should be
+        NSLayoutConstraint.activate([
+            self.customTabBar.leadingAnchor.constraint(equalTo: tabBar.leadingAnchor),
+            self.customTabBar.trailingAnchor.constraint(equalTo: tabBar.trailingAnchor),
+            self.customTabBar.widthAnchor.constraint(equalToConstant: tabBar.frame.width),
+            self.customTabBar.heightAnchor.constraint(equalToConstant: 67), // Fixed height of 67pts for nav menu. This value does not include the curved edge insets in devices >= iPhone X
+            self.customTabBar.bottomAnchor.constraint(equalTo: tabBar.bottomAnchor)
+        ])
+        
+        // Custom tab bar has been created, we want to react to taps on TabNavigationMenu items the moment they happen
+        self.customTabBar.itemTapped = { index in
+            self.selectedIndex = index
+            if index == 0 {
+                self.showFloatingButtons()
+            } else {
+                self.hideFloatingButtons()
+            }
+        }
+        
+        var viewControllers = [UIViewController]()
+        
```

Comment:
This empty line space can be removed. The preceding line and the next set of lines are very much related.

Reply:
resolved in 561e660